mne_bids/write.py

```diff
@@ -338,7 +338,28 @@ def _events_json(fname, overwrite=False):
         Whether to overwrite the output file if it exists.
     """
     new_data = {
-        "sample": {"Description": "The event onset time in number of sampling points."},
+        "onset": {
+            "Requirement Level": "REQUIRED",
+            "Data type": "number",
+            "Description": (
+                "Onset (in seconds) of the event from the beginning of the first data"
+                "point. Negative onsets account for events before the first stored "
+                "data point."
+            ),
+        },
+        "duration": {
+            "Requirement Level": "REQUIRED",
+            "Data type": "number or 'n/a'",
+            "Description": (
+                "Duration of the event in seconds from onset. "
+                "Must be zero, positive, or 'n/a' if unavailable. "
+                "A zero value indicates an impulse event. "
+            ),
+        },
+        "sample": {
+            "Data type": "number",
+            "Description": ("The event onset time in number of sampling points."),
```

Comment:
Perhaps you could double-check this, but I think our "sample" column is zero indexed.

Perhaps you could say so in the Description here.

- xref https://github.com/bids-standard/bids-specification/pull/1457

Reply:
Not sure if I understand the discussion correctly, but as I read it, the user can decide how a sample is used. I now added that the first value is supposed to be 0.